### ISSUE TITLE
Locally installed Chef server

### ISSUE THREAD

**Issue by buep** [Praqma/study-succotash#23]
**Timeboxed research**

On Chef.io the free version only goes for 5 nodes. I have a customer that needs more then 5 nodes now, so as I far as I can understand we need a local server on-premise that support up to 25 nodes with full feature-set or move to use Chef solo.
We won't yet pay for more licenses in the cloud.

I also have the option to spin up a Chef server in Azure, but there is a lot of manual steps requires and want to automate it, so I'm thinking to use Docker.

These might be good candidates after some short research:
* https://hub.docker.com/r/xmik/chef-server-docker/
* https://hub.docker.com/r/stocksoftware/chef-server/

### (Must have) Try one of the images
Chose of of them and spin it up - does it support all the same things as the online chef.io for 5 nodes free license?
How does it compare ? Anything missing?
How do we create users?

### (Could have) If the first one isn't good, try the next
Same as for the first one

###  (Must Have) How to migrate from the cloud version to on premise self-hosted in docker?

* How do we for existing nodes managed by the cloud server move them to be managed by the local one?
* can Chef vault be used with the new server? how to move existing Chef vault?

### (Must Have)

What is the exact steps I need to do to get the chef-server running in a container, and migrate our existing setup to that one?

**Comment by mortench3000** [comment]
I guess that the timebox is maxed out, so here is what I have so far:

I chose to work with the [stocksoftware](https://hub.docker.com/r/stocksoftware/chef-server/) image
It's based off a Chef-Server v. 11 which is one version short of the latest.

Used Run command from instructions, and the server came up within a few minutes...
It uses a selfsigned certificate which would have to be replaced in a 'real' environment.
I had no problems accessing the server through web, but the knife utility however, would not accept to connect unless the url was set to 127.0.0.1.
Inside the container, the cert is located in /var/opt/chef-server/nginx/ca
Tried to experiment with the PUBLIC_URL environment variable as the instructions mentions, but had no luck - Could be lack of Docker experience though...
Using this run command instead
 `docker run -it -e PUBLIC_URL='api.chef.local' --name chef_server -p 80:80 -p 443:443 stocksoftware/chef-server`
just generated a weird looking cert, and then I couldn't get in with web.
Also, I have only chef managed servers in VM's, and I could not, within the time, figure out how to get the servers to connect to the containerized chef-server running on 127.0.0.1. Hope the problem is clear...

**Procedure for migration once the chef-server is up and running:**
Withing the chef-repo:
Update the .chef directory with the following:
 - Remove existing .pem files from old hosted server
 - Add chef-validator.pem and [user].pem from the new server

Upload the cookbook(s) with dependencies:
`berks upload`
Add the nodes:
`knife node add [node]`
Add the runlist to the nodes:
`knife node run_list set [node] recipe[cookbook]`
Add the vault and tie it to users and nodes
`?`
Run the recipies on all nodes (may need to re-bootstrap to get the new keys downloaded to the nodes)
